galsim/chromatic.py

```diff
@@ -19,45 +19,89 @@
+galsim.Bandpass instance).
 
 Possible uses include galaxies with color gradients, automatically drawing a given galaxy through
 different filters, or implementing wavelength-dependent point spread functions.
 """
 
-from future.utils import iteritems
 import numpy as np
-import copy
 
 import galsim
-from functools import reduce
+
 
 class ChromaticObject(object):
     """Base class for defining wavelength-dependent objects.
 
-    This class primarily serves as the base class for chromatic subclasses, including Chromatic,
-    ChromaticSum, and ChromaticConvolution.  See the docstrings for these classes for more details.
+    This class primarily serves as the base class for chromatic subclasses.  See the docstrings for
+    subclasses for more details.
 
     Initialization
     --------------
 
-    A ChromaticObject can also be instantiated directly from an existing GSObject.
-    In this case, the newly created ChromaticObject will act nearly the same way the original
-    GSObject works, except that it has access to the ChromaticObject methods described below;
-    especially expand(), dilate() and shift().
-
-    @param gsobj  The GSObject to be chromaticized.
+    A ChromaticObject can be instantiated from an existing GSObject by applying a
+    wavelength-dependent transformation.  E.g.,
+
+    >>> gsobj = galsim.Gaussian(fwhm=1)
+    >>> chrom_obj = gsobj.dilate(lambda wave: 1.01**wave)
+
+    In this and similar cases, the univariate object passed to the transformation method (`dilate`
+    above) is a python callable that accepts wavelength in nanometers and returns whatever type the
+    transformation method normally accepts (so an int or float above).
+
+    Another way to instantiate a ChromaticObject from a GSObject is to multiply by an SED.  This can
+    be useful to consistently generate the same galaxy observed through different filters, or, with
+    ChromaticSum, to construct multi-component galaxies, each component with a different SED.  For
+    example, a bulge+disk galaxy could be constructed:
+
+    >>> bulge_SED = user_function_to_get_bulge_spectrum()
+    >>> disk_SED = user_function_to_get_disk_spectrum()
+    >>> bulge_mono = galsim.DeVaucouleurs(half_light_radius=1.0)
+    >>> disk_mono = galsim.Exponential(half_light_radius=2.0)
+    >>> bulge = bulge_mono * bulge_SED
+    >>> disk = disk_mono * disk_SED
+    >>> gal = bulge + disk
+
+    The SEDs above describe the flux density in photons/nm/cm^2/s of an object, possibly normalized
+    with either the sed.withFlux(bandpass) or sed.withMagnitude(bandpass) methods (see the
+    docstrings in the SED class for details about these and other normalization options).  Note that
+    the `flux` attribute of the multiplied GSObject is accounted for in the newly constructed
+    ChromaticObject's normalization.  I.e., the following are equivalent:
+
+    >>> chrom_obj = (sed * 3.0) * gsobj
+    >>> chrom_obj2 = sed * (gsobj * 3.0)
```

Comment:
Also realized that if we're bringing back `chrom = ChromaticObject(gsobj)`, we need to decide the dimensions of `chrom`.  I think dimensionless makes the most sense here too, so I added the following to the `ChromaticObject` docstring:

```
    One caveat to creating ChromaticObjects directly from GSObjects like this is that even though
    the source GSObject instance has flux units in photons/s/cm^2, then newly formed ChromaticObject
    will be interpreted as dimensionless, i.e., it will have a dimensionless SED.  See below for
    more discussion about the dimensions of ChromaticObjects.
```